src/citrine/ara/variables.py

```diff
@@ -467,9 +467,138 @@ def __init__(self, *,
                  headers: List[str],
                  attribute_template: LinkByUID,
                  process_templates: List[LinkByUID],
-                 attribute_constraints: List[List[Union[LinkByUID, BaseBounds]]] = None):
+                 attribute_constraints: Optional[List[List[Union[LinkByUID, BaseBounds]]]] = None):
         self.name = name
         self.headers = headers
         self.attribute_template = attribute_template
         self.process_templates = process_templates
         self.attribute_constraints = attribute_constraints
+
+
+class IngredientIdentifierInOutput(Serializable['IngredientIdentifierInOutput'], Variable):
+    """[ALPHA] Ingredient quantity in the trunk of a material history tree.
+
+    The search for an ingredient quantity starts at the root of the material history tree
```

Comment:
Not your issue, but we are officially updating all docs to refer to `terminal materials` instead of roots, since there is conceptual ambiguity around roots not being the originating materials.  I haven't gotten started on the bit of work yet.

https://citrine.atlassian.net/browse/PLA-4198

Reply:
Good to know. I'll update the docs. so there will be two fewer places we need to update in the future.